presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -425,7 +431,7 @@ private PrestoSparkTaskSourceRdd createTaskSourcesRdd(
             for (int splitIndex = 0; splitIndex < splits.size(); splitIndex++) {
                 result.put(splitIndex % initialPartitionCount, splits.get(splitIndex));
             }
-            return result.build();
+            return result;
```

Comment:
Is building the map causing OOM? The built immutable map might be more space efficient?

Reply:
It's not the immutable map that increases the memory footprint, but the inability to remove serialized items from the map. I'm trying to avoid retaining both, serialized and deserialized versions in memory at the same time.